Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5680,14 +5680,33 @@ LayoutBoxExtent RenderBox::scrollPaddingForViewportRect(const LayoutRect& viewpo
 
 LayoutUnit synthesizedBaseline(const RenderBox& box, const RenderStyle& parentStyle, LineDirectionMode direction, BaselineSynthesisEdge edge)
 {
+    auto baselineType = [&] {
+        // https://drafts.csswg.org/css-inline-3/#alignment-baseline-property
+        // https://drafts.csswg.org/css-inline-3/#dominant-baseline-property
+
+        auto isHorizontalWritingMode = parentStyle.isHorizontalWritingMode();
+        auto textOrientation = parentStyle.textOrientation();
+
+        if (isHorizontalWritingMode || (!isHorizontalWritingMode && textOrientation == TextOrientation::Sideways))
+            return FontBaseline::AlphabeticBaseline;
+        if (!isHorizontalWritingMode && (textOrientation == TextOrientation::Upright || textOrientation == TextOrientation::Mixed))
```

Comment:
this too as horizontal mode is already handled.
`if (textOrientation == TextOrientation::Upright || textOrientation == TextOrientation::Mixed)`

Reply:
or you could turn this into something like this

```
if (isHorizontalWritingMode)
  return FontBaseline::AlphabeticBaseline;
else {
  switch (parentStyle.textOrientation() {
    case TextOrientation::Sideways:
      return FontBaseline::AlphabeticBaseline;
   ...
}
```
to avoid consulting parentStyle.textOrientation() in the common case of horizontal writing mode, but since this is certainly not a hot codepath, your original approach is okay too.

Reply:
Fixed